### PR TITLE
Remove `Timestamps` and `Conflicts` from `FpcPayload`

### PR DESCRIPTION
**Pull request by pvdrz**
I think those two types were something we used before deciding to use `VecPrefix` everywhere.

### CODE REVIEW

File: bee-message/tests/fpc_payload.rs [hunk 1]
```diff
@@ -130,26 +119,26 @@ fn unpack_valid() {
 }
 
 #[test]
-fn packable_round_trip() {
+fn round_trip() {
```
Comment (Adam-Gleave) [inline bee-message/tests/fpc_payload.rs]: I would still keep this as `packable_round_trip`, I think the rest of the tests are named that.

File: bee-message/src/payload/fpc/mod.rs [hunk 2]
```diff
@@ -3,25 +3,55 @@
...
+pub use timestamp::Timestamp;
 
-use bee_packable::Packable;
+use crate::{
+    payload::{transaction::TransactionId, MessagePayload, PAYLOAD_LENGTH_MAX},
+    MessageId, MessageUnpackError, ValidationError,
+};
+
+use bee_packable::{error::UnpackPrefixError, BoundedU32, Packable, VecPrefix};
+
```
Comment (Adam-Gleave) [inline bee-message/src/payload/fpc/mod.rs]: Order! :trollface:

File: bee-message/tests/fpc_payload.rs (hunk 1 above)
Comment (thibault-martinez) [reply bee-message/tests/fpc_payload.rs]: I agree, let's keep it for consistency

**Review by Adam-Gleave — DISMISSED** [review]

File: bee-message/src/payload/fpc/conflict.rs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+use crate::payload::transaction::TransactionId;
+
+use bee_packable::Packable;
+
+/// Describes a vote in a given round for a transaction conflict.
+#[derive(Clone, Debug, Eq, PartialEq, Packable)]
+#[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
+pub struct Conflict {
+    /// ID of the conflicting transaction.
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/conflict.rs]:
```suggestion
    /// Identifier of the conflicting transaction.
```

File: bee-message/src/payload/fpc/conflict.rs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    /// Creates a new [`Conflict`].
+    pub fn new(transaction_id: TransactionId, opinion: u8, round: u8) -> Self {
+        Self {
+            transaction_id,
+            opinion,
+            round,
+        }
+    }
+
+    /// Returns the ID of the conflicting transaction.
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/conflict.rs]:
```suggestion
    /// Returns the identifier of the conflicting transaction.
```

File: bee-message/src/payload/fpc/timestamp.rs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+use crate::MessageId;
+
+use bee_packable::Packable;
+
+/// Describes a vote in a given round for a message timestamp.
+#[derive(Clone, Debug, Eq, PartialEq, Packable)]
+#[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
+pub struct Timestamp {
+    /// ID of the message that contains the timestamp.
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/timestamp.rs]:
```suggestion
    /// Identifier of the message that contains the timestamp.
```

File: bee-message/src/payload/fpc/conflict.rs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+use bee_packable::Packable;
+
+/// Describes a vote in a given round for a transaction conflict.
+#[derive(Clone, Debug, Eq, PartialEq, Packable)]
+#[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
+pub struct Conflict {
+    /// ID of the conflicting transaction.
+    transaction_id: TransactionId,
+    /// The nodes opinion value in a given round.
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/conflict.rs]: Just checking, is this supposed to be the opinion of a specific node or several nodes ?
`nodes` vs `node's` ?

File: bee-message/src/payload/fpc/timestamp.rs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    /// Creates a new [`Timestamp`].
+    pub fn new(message_id: MessageId, opinion: u8, round: u8) -> Self {
+        Self {
+            message_id,
+            opinion,
+            round,
+        }
+    }
+
+    /// Returns the ID of the message that contains the timestamp.
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/timestamp.rs]:
```suggestion
    /// Returns the identifier of the message that contains the timestamp.
```

File: bee-message/src/payload/fpc/mod.rs [hunk 8]
```diff
@@ -3,25 +3,54 @@
...
-pub use timestamps::{Timestamp, Timestamps};
+pub use conflict::Conflict;
+pub use timestamp::Timestamp;
 
-use crate::{payload::MessagePayload, MessageUnpackError, ValidationError};
+use crate::{
+    payload::{transaction::TransactionId, MessagePayload, PAYLOAD_LENGTH_MAX},
+    MessageId, MessageUnpackError, ValidationError,
+};
+use bee_packable::{error::UnpackPrefixError, BoundedU32, Packable, VecPrefix};
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/mod.rs]:
```suggestion

use bee_packable::{error::UnpackPrefixError, BoundedU32, Packable, VecPrefix};
```

File: bee-message/src/payload/fpc/mod.rs [hunk 9]
```diff
@@ -3,25 +3,54 @@
...
 
-use bee_packable::Packable;
+use core::convert::Infallible;
+
+/// No [`Vec`] max length specified, so use [`PAYLOAD_LENGTH_MAX`] / length of [`Conflict`].
+const PREFIXED_CONFLICTS_LENGTH_MAX: u32 =
+    PAYLOAD_LENGTH_MAX / (TransactionId::LENGTH + 2 * core::mem::size_of::<u8>()) as u32;
+
+/// No [`Vec`] max length specified, so use [`PAYLOAD_LENGTH_MAX`] / length of
+/// [`Conflict`](crate::payload::fpc::Conflict).
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/mod.rs]: Conflict is on the scope, no need for the parenthesis

File: bee-message/src/payload/fpc/mod.rs [hunk 10]
```diff
@@ -3,25 +3,54 @@
...
+const PREFIXED_TIMESTAMPS_LENGTH_MAX: u32 =
+    PAYLOAD_LENGTH_MAX / (MessageId::LENGTH + 2 * core::mem::size_of::<u8>()) as u32;
+
+fn unpack_prefix_to_timestamp_validation_error(error: UnpackPrefixError<Infallible>) -> ValidationError {
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/mod.rs]: I am being extremely picky now but can you put this function after the conflict one please ?

File: bee-message/src/payload/fpc/mod.rs [hunk 11]
```diff
@@ -60,13 +89,19 @@ impl FpcPayloadBuilder {
     }
 
     /// Adds a collection of conflicts to the [`FpcPayloadBuilder`].
-    pub fn with_conflicts(mut self, conflicts: Conflicts) -> Self {
+    pub fn with_conflicts(
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/mod.rs]: I feel like this is a not really handy API for the user. Maybe we should take Vec here and try to convert them to the appropriate VecPrefix when calling `finish` and propagating errors ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: bee-message/src/payload/fpc/conflict.rs (hunk 6 above)
Comment (Adam-Gleave) [reply bee-message/src/payload/fpc/conflict.rs]: It should be `node's`.

File: bee-message/src/payload/fpc/timestamp.rs [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+            round,
+        }
+    }
+
+    /// Returns the identifier of the message that contains the timestamp.
+    pub fn message_id(&self) -> &MessageId {
+        &self.message_id
+    }
+
+    /// Returns the nodes opinion value in a given round.
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/timestamp.rs]:
```suggestion
    /// Returns the node's opinion value in a given round.
```

File: bee-message/src/payload/fpc/conflict.rs [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+            round,
+        }
+    }
+
+    /// Returns the identifier of the conflicting transaction.
+    pub fn transaction_id(&self) -> &TransactionId {
+        &self.transaction_id
+    }
+
+    /// Returns the nodes opinion value in a given round.
```
Comment (thibault-martinez) [inline bee-message/src/payload/fpc/conflict.rs]:
```suggestion
    /// Returns the node's opinion value in a given round.
```

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**